### ISSUE TITLE
Let races be deleted

### ISSUE THREAD

**Issue by glacials** [glacials/splits-io#740]
Many users get confused about how to remove a race that they don't want polluting their profile etc., we should allow them to be deleted. It needs to be a soft deletion (a `deleted_at` column) because of the way we use race UUIDs in URLs.

**Comment by BatedUrGonnaDie** [comment]
As a compromise for this, any race that starts cannot be deleted?  That seems to make it so you can't abuse it while also giving users the impression that they can test something out and still remove the race.

**Comment by glacials** [comment]
Some of the reports are from races that were started as a test. Maybe any race with >1 non-ghost participant cannot be deleted?

**Comment by BatedUrGonnaDie** [comment]
I thought you couldn't start races unless there were 2 real entrants?

**Comment by glacials** [comment]
Nope, ghosts count: https://github.com/glacials/splits-io/blob/main/app/models/race.rb#L96-L119

**Comment by BatedUrGonnaDie** [comment]
Ah, then I agree probably if there is >1 non-ghost it shouldn't be delete-able (once started).
